pkg/mark/macro/macro.go

```diff
@@ -134,13 +134,36 @@ func ExtractMacros(
 				macro Macro
 			)
 
-			macro.Template, err = includes.LoadTemplate(base, template, templates)
-			if err != nil {
-				err = karma.Format(err, "unable to load template")
+			if strings.HasPrefix(template, "#") {
+				cfg := map[string]interface{}{}
 
-				return nil
+				err = yaml.Unmarshal([]byte(config), &cfg)
+				if err != nil {
+					err = karma.Format(
+						err,
+						"unable to unmarshal macros config template",
+					)
+					return nil
```

Comment:
should not we return the error here and below? karma.Format only adds a context-like message to an error.

Reply:
I copied the error handing from the error handling already present there. You must know: I'm just a go-beginner and mostly learn by looking at other's code. So I assumed the error handling here https://github.com/kovetskiy/mark/blob/4c812741ac590eb153193fe2c6d228e6845974f7/pkg/mark/macro/macro.go#L138-L142 is correct.

Reply:
Ah, you are actually right.